src/mon/HealthMonitor.cc

```diff
@@ -682,6 +683,27 @@ bool HealthMonitor::check_leader_health()
 
   health_check_map_t next;
 
+ // DAE_VERSION_INCORRECT
+  std::map<int, std::pair<int, string>> wrong;
+  double timeout = 600;
+  if (mon->check_daemon_versions(wrong)){
+    time_t current = time(NULL);
+    if ((timeout < difftime(current, start))){
+      ostringstream ss, ds;
+      ss << "There are daemon running incorrect version of ceph \n";
```

Comment:
```suggestion
      ss << "There are daemons running incorrect version of ceph \n";
```

another possible improvement would be to print is/are based on the size of wrong